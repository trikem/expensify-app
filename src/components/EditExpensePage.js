import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { editExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.editExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };
    onRemove = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    };
    render() {
        return (
            <div>
                <ExpenseForm
                    onSubmit={this.onSubmit} 
                    expense={this.props.expense}
                />
                <button onClick={this.onRemove}>Remove</button>
            </div>
        );
    };
}

const mapStateToProps = (state, props) => ({ 
    expense: state.expenses.find((expense) => expense.id === props.match.params.id) 
});

const mapDispatchToProps = (dispatch, props) => ({
    editExpense: (id, expense) => dispatch(editExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);