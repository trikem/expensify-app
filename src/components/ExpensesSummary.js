import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => (
    <div>
    {
        props.expenses.length !== 0 && ( 
            <p>Viewing {props.expenses.length} expenses totaling {numeral(getExpensesTotal(props.expenses)/100).format('$0,0.00')}</p> 
        )
    }
    </div>
);

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        
        expenses: visibleExpenses     
    }
}

export default connect(mapStateToProps)(ExpensesSummary);