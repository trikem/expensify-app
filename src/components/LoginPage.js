import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
  <div class="box-layout">
    <div class="box-layout__box">
      <h1 className="box-layout__title">Expensify</h1>
      <p>Its' time</p>
      <button className="button" onClick={startLogin}>Login with Google</button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);