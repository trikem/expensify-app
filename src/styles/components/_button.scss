.button {
  background: $light-blue;
  padding: $s-size;
  border: none;
  color: white;
  font-size: $font-size-large;
  font-weight: 300;
  display: inline-block;
  text-decoration: none;
  line-height: 1;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888888;
}