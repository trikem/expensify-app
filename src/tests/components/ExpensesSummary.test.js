import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render exp summary', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render exp summary with empty message', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
});