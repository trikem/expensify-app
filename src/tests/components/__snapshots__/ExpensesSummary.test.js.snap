// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render exp summary 1`] = `
<div>
  <h1>
    Viewing 
     
    expenses
     totaling 
    $0.00
  </h1>
</div>
`;

exports[`should render exp summary with empty message 1`] = `
<div>
  <h1>
    Viewing 
     
    expenses
     totaling 
    $0.00
  </h1>
</div>
`;
